File: crafting_with_solid_principles/code/java/1-s.java
public class UserService {
  private final String smtplogin;
  private Database _database;

  public UserService(String smtplogin) {
    this.smtplogin = smtplogin;
    _database = Database.getInstance();
  }

  public void register(String email, String password) {
    if (!email.contains("@"))
      throw new ValidationException("Email is not an email!");
    User user = new User(email, password);
    _database.save(user);
    SmtpClient _smtpClient = SmtpClient.connect(Smtp.getConf(), smtplogin);
    _smtpClient.send(new MailMessage("mysite@nowhere.com", email) {{
      subject = "HEllo fool!";
    }});
  }
}

File: crafting_with_solid_principles/code/python/1-s.py

class UserService:
    def __init__(self, smtplogin):
        self.smtplogin = smtplogin
        self._database = Database.getInstance()

    def register(self, email, password):
        if not email.contains("@"):
            raise "Email is not an email!"
        user = User(email, password);
        self._database.save(user);
        smtpClient = SmtpClient.connect(Smtp.getConf(), self.smtplogin);
        smtpClient.send(MailMessage("mysite@nowhere.com", email, subject = "HEllo fool!")
